### PR TITLE
OSSM-6774 Ensure that metrics don't require a separate kube & discovery client

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

File: pkg/metrics/metrics.go [hunk 1]
```diff
@@ -42,13 +42,19 @@ const (
...
 		return nil, fmt.Errorf("failed to create new client: %w", err)
 	}
+	return CreateMetricsService0(ctx, client, servicePorts)
+}
+
+// CreateMetricsService creates a Kubernetes Service to expose the passed metrics
+// port(s) with the given name(s).
+func CreateMetricsService0(ctx context.Context, client crclient.Client, servicePorts []v1.ServicePort) (*v1.Service, error) {
+	if len(servicePorts) < 1 {
+		return nil, fmt.Errorf("failed to create metrics Serice; service ports were empty")
```
Comment (yannuil) [inline pkg/metrics/metrics.go]: I think there is a typo here `Serice`

**Review by eoinfennessy — APPROVED** [review]

**Review by yannuil — APPROVED** [review]
LGTM

**Comment by luksa** [comment]
Merging manually, because we don't have automation in this repo.

File: pkg/metrics/metrics.go (hunk 1 above)
Comment (luksa) [reply pkg/metrics/metrics.go]: Fixed. Thanks.

**Pull request merged**